test/extended/router/idle.go

```diff
@@ -25,36 +26,54 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
 	var (
 		configPath = exutil.FixturePath("testdata", "router", "router-idle.yaml")
 		oc         = exutil.NewCLI("router-idling")
+
+		// The state of all the resources in the test
+		// namespace. Printed if the test fails.
+		resourceState string
 	)
 
 	// this hook must be registered before the framework namespace teardown
 	// hook
 	g.AfterEach(func() {
 		if g.CurrentGinkgoTestDescription().Failed {
 			exutil.DumpPodLogsStartingWithInNamespace("router", "openshift-ingress", oc.AsAdmin())
+			e2e.Logf("resource state:\n%s\n", resourceState)
 		}
 	})
 
 	g.Describe("The HAProxy router", func() {
 		g.It("should be able to connect to a service that is idled because a GET on the route will unidle it", func() {
 			// timeout for kube GET polling operations
-			timeout := 3 * time.Minute
+			timeout := 5 * time.Minute
 
-			// timout for GET requests
-			urlTimeout := 1 * time.Minute
+			// urlTimeout for HTTP GET requests
+			urlTimeout := 2 * time.Minute
 
 			g.By(fmt.Sprintf("creating test fixture from a config file %q", configPath))
 			err := oc.Run("new-app").Args("-f", configPath).Execute()
 			o.Expect(err).NotTo(o.HaveOccurred())
 
-			urlTester := url.NewTester(oc.AdminKubeClient(), oc.Namespace()).WithErrorPassthrough(true)
-			defer urlTester.Close()
+			defer func() {
+				// Accumulate current state of all
+				// resources in the test namespace. Do
+				// this here as Namespace.Name is nil
+				// in AfterEach() when the test has
+				// failed.
+				resourceState, _ = oc.Run("get").Args("all", "-o", "yaml").Output()
+			}()
+
+			g.By("Waiting for pods to be running")
+			err = waitForRunningPods(oc, 1, exutil.ParseLabelsOrDie("app=idle-test"), urlTimeout)
```

Comment:
Use `timeout` here instead of `urlTimeout`?

Reply:
I have one reproducer in local testing today and will fiddle with all the timeouts we're using. Think they're all just too low at the moment.

Reply:
Unequivocally, we shouldn't need second test for "pods are running" because a 200 GET is effectively the same test. I'll remove that - just wanted more evidence about where the flakes occur. Seems it can be in various places at the moment, not just one aspect of the test.